src/fhirtypes/common.ts

```diff
@@ -171,14 +194,34 @@ export function createUsefulSlices(
               current[key][effectiveIndex] == null
             ) {
               current[key][effectiveIndex] = {};
+              if (underscored) {
+                current[pathPart.base][effectiveIndex] = {};
+              } else if (pathPart.primitive) {
+                current[`_${pathPart.base}`][effectiveIndex] = {};
+              }
             } else if (j >= current[key].length) {
               if (sliceName) {
                 // _sliceName is used to later differentiate which slice an element represents
                 current[key].push({ _sliceName: sliceName });
+                if (underscored) {
+                  current[pathPart.base][current[key].length - 1] = { _sliceName: sliceName };
+                } else if (pathPart.primitive) {
+                  current[`_${pathPart.base}`][current[key].length - 1] = { _sliceName: sliceName };
+                }
               } else if (j === effectiveIndex) {
                 current[key].push({});
+                if (underscored) {
+                  current[pathPart.base][current[key].length - 1] = null;
```

Comment:
In several of the situations above, when it is `underscored`, we set the `pathPart.base` value to `{}`, but here we set it to `null`. Is this difference intentional?